pkg/controller/controller.go

```diff
@@ -923,8 +940,16 @@ func (p *csiProvisioner) Provision(ctx context.Context, options controller.Provi
 	return pv, controller.ProvisioningFinished, nil
 }
 
-func (p *csiProvisioner) setCloneFinalizer(ctx context.Context, pvc *v1.PersistentVolumeClaim) error {
-	claim, err := p.claimLister.PersistentVolumeClaims(pvc.Namespace).Get(pvc.Spec.DataSource.Name)
+func (p *csiProvisioner) setCloneFinalizer(ctx context.Context, pvc *v1.PersistentVolumeClaim, dataSource *v1.ObjectReference) error {
+	var claim *v1.PersistentVolumeClaim
+	var err error
+
+	if len(dataSource.Namespace) > 0 {
```

Comment:
Can `dataSource` also set the namespace field so we don't need this check?

Reply:
It is possible.